src/preflight/handler.js

```diff
@@ -66,169 +67,141 @@ export const preflightAudit = async (context) => {
   const { urls, step = AUDIT_STEP_IDENTIFY } = jobMetadata.payload;
   const normalizedStep = step.toLowerCase();
 
-  log.info(`[preflight-audit] site: ${site.getId()}. Preflight audit started for jobId: ${job.getId()}`);
-  log.info(`[preflight-audit] site: ${site.getId()}. Step: ${normalizedStep}`);
+  log.info(`[preflight-audit] site: ${site.getId()}. Preflight audit started for jobId: ${job.getId()} and step: ${normalizedStep}`);
 
   if (job.getStatus() !== AsyncJob.Status.IN_PROGRESS) {
     throw new Error(`[preflight-audit] site: ${site.getId()}. Job not in progress for jobId: ${job.getId()}. Status: ${job.getStatus()}`);
   }
 
-  const result = {
-    audits: [],
-  };
-
-  result.audits.push({
-    name: 'canonical',
-    type: 'seo',
-    opportunities: [],
-  });
-
-  const storagePathSet = new Set(urls.map((url) => {
-    const pathname = new URL(url).pathname.replace(/\/$/, '');
-    return `scrapes/${site.getId()}${pathname}/scrape.json`;
+  const pageAuthToken = await retrievePageAuthentication(site, context);
+  const baseURL = new URL(urls[0]).origin;
+  const authHeader = { headers: { Authorization: `token ${pageAuthToken}` } };
+
+  // Initialize results
+  const result = urls.map((url) => ({
+    pageUrl: url,
+    step: normalizedStep,
+    audits: AUDIT_NAMES.map((name) => ({ name, type: 'seo', opportunities: [] })),
   }));
+  const resultMap = new Map(result.map((r) => [r.pageUrl, r]));
+
+  // Canonical checks
+  const canonicalResults = await Promise.all(
+    urls.map(async (url) => {
+      const { canonicalUrl, checks: tagChecks } = await validateCanonicalTag(url, log, authHeader);
```

Comment:
It would be really beneficial to annotate the returned values consistently throughout the code.

Reply:
@dzehnder , ^^^ I meant adding code annotations to the returned responses - not unifying audit responses at this stage